contracts/balancer/IndexPool.sol

```diff
@@ -1324,8 +1202,10 @@ contract IndexPool is BToken, BMath, IIndexPool {
       denorm = uint96(badd(oldWeight, maxDiff));
       diff = maxDiff;
     }
-    _totalWeight = badd(_totalWeight, diff);
-    require(_totalWeight <= MAX_TOTAL_WEIGHT, "ERR_MAX_TOTAL_WEIGHT");
+    // If new total weight exceeds the maximum, do not update
+    uint256 newTotalWeight = badd(_totalWeight, diff);
+    if (newTotalWeight > MAX_TOTAL_WEIGHT) return;
```

Comment:
Is it fine to have it pass the max under the hood without the state being aligned?

Reply:
It won't update the total here because the actual state change that modifies `record.denorm` in storage only happens below this condition, in the line:
```
_records[token].denorm = denorm;
```